### ISSUE TITLE
PieChart "setState('default')" does not reset "isActive"

### ISSUE THREAD

**Issue by sassomedia** [amcharts/amcharts4#677]
When PieChart slices are activated, then reset with `slice.setState('default')`, `slice.isActive` is not reset to `false`. This is causing the slices to activate again on next rollover.

Issue shown [here](https://jsfiddle.net/sassomedia/bognsLvf/):
1. Click on a few slices
2. Click first button
3. Rollover chart, slices activate again

Expected behavior: click on second button and rollover chart

**Comment by martynasma** [comment]
I'm not entirely sure this should be the case.

States are about applying certain set of property values.

`isActive` is about funcationality/logic.

We will consider it, but not sure which way we will go, yet.

For you, it's probably best if you also set `isActive = false` in the same place where you call `setState('default')`.

**Comment by sassomedia** [comment]
I think I get your point, what probably could need more clarification are the [docs](https://www.amcharts.com/docs/v4/concepts/states/#_default_state):
> A "default" state is a special state which holds initial values for element's properties and filters.

The reason why I would expect a reset is because the slice in its original position/shape has the property `isActive` set to `false`, unless specifically set to `pulled` or `true` via config. I assumed that would be considered the default state. - While I don't mind specifying `isActive` separately, to how many other properties would this apply to as well? The docs probably should detail which properties the state actually encompasses.

I guess what I'm looking for is a simple way to reset a chart to its original look/state after any number of changes have been made to it. `setState` seemed like a great way to do this as it does handle position, size/shape and color.

**Comment by martynasma** [comment]
I totally agree that state definition could use some clarification.

Let me explain it here (I will think of how we can put that in docs as well).

"Default state" is not identical set of properties that object had at its birth.

It's rather set of values for those properties that were overridden by another state.

So, for example, if your slice has a `fill` set to red, and a hover state that overrides `fill` to green. When you go back to default state, the `fill` will be reset back to red.

If, at some point between birth and going back to default state slice's `isActive` (or any other property for that matter) was changed by some code, it is not considered a state-related matter, and therefore not reset.

**So, bottom line, default state resets the values for only those properties that were overridden by another state.**

That said, if you would add `isActive` to the active state's `properties`, then technically, it should reset back to `false` when you go back to default state.

I hope that makes sense.

**Comment by martynasma** [comment]
I'm closing this, but please feel free to post any follow up questions you may have.

**Comment by sassomedia** [comment]
@martynasma Ok, makes sense. Thanks!
